src/bundle/bundleRouter.ts

```diff
@@ -109,8 +127,20 @@ export function createBundleRouter(props: IBundleRouteProps): IBundleRouter {
           if (!cssBundle) createCSSBundle();
           cssBundle.source.modules.push(module);
         } else if (module.pkg.type === PackageType.EXTERNAL_PACKAGE && hasVendorConfig) {
-          if (!vendorBundle) createVendorBundle();
-          vendorBundle.source.modules.push(module);
+          let isMappedBundle = false;
+          if (hasMappings) {
+            for (const mapping of mappings) {
+              if (mapping.regexp.test(module.pkg.publicName)) {
+                createSubVendorBundle(module, mapping);
+                isMappedBundle = true;
+                break;
+              }
+            }
+          }
+          if (!isMappedBundle) {
```

Comment:
Also, do you think you can write a few tests? There are plenty of integration tests.

Reply:
Absolutelly! As soon I have some time to spare I'll do it!